### PR TITLE
[QUIZ-145] feat: api get member in group, invite member

### PR DESCRIPTION
**Pull request by duchuyyyy**
# Pull Request types

- 🐛 Bug Fix
- 👍 Improvement
- ✨ New Feature
- 🎉 Release
- ♻️ Refactoring
- 🚿 Refactoring (Remove functions)
- 💚 Refactoring (Test or CI)
- 👕 Fix (Lint)
- 🆙 Update (Dependency packages)
- 🚀 Improvement (Performance)
- 👮 Improvement (Security)
- 🔒 Restriction (Restrict new feature)
- 👾 Others

# Description

# Checklist

- [ ] 1. Build successfully on local
- [ ] 2. Format all code files included in the PR
- [ ] 3. No more unconfirmed specs in the PR
- [ ] 4. Tested the normal case and error cases following specs
- [ ] 5. Is there a refactor to the old code? If yes, have you tested the old function and noted the impact yet?
- [ ] 6. Loops have a set length and correct termination conditions
- [ ] 7. There aren't any redundant or duplicate codes in PR (clean code)
- [ ] 8. There aren't any hard-coded in the code in PR
- [ ] 9. Optimization logic handling and optimizations SQL (for example, don't get redundant data or add an index to optimize performance, query only necessary fields...)
- [ ] 10. Update issue status and add PR link to the issue

# Evidence
(Screenshot or Video)

# Discussion

### CODE REVIEW

File: src/modules/group/group.controller.ts [hunk 1]
```diff
@@ -31,7 +43,59 @@ export class GroupController {
...
+    name: 'groupId',
+    description: 'The UUID of the Group',
+    type: 'string',
+  })
+  @Get('groups/:groupId/members')
+  async getMemberInGroup(
+    @Param('groupId', ValidateUuid) groupId: Uuid,
+    @CurrentUser('id') userId: Uuid,
+  ) {
+    return this.groupService.getMemberInGroup(groupId, userId);
```
Comment (LeTatTuan) [inline src/modules/group/group.controller.ts:83]: - mấy hàm ni nên để object khi truyền vào á

File: src/modules/group/group.controller.ts [hunk 2]
```diff
@@ -31,7 +43,59 @@ export class GroupController {
...
+  @ApiAuth({
+    summary: 'Member joins group',
+  })
+  @ApiParam({
+    name: 'groupId',
+    description: 'The UUID of the Group',
+    type: 'string',
+  })
+  @Post('group/:groupId/members/joins')
+  async joinRoom(
```
Comment (LeTatTuan) [inline src/modules/group/group.controller.ts:95]: - join group m

File: src/modules/group/group.service.ts [hunk 3]
```diff
@@ -29,20 +37,77 @@ export class GroupService {
...
+  ) {
+    const isUserHasRoleHostInGroup =
+      await this.memberInGroupService.isUserHasRoleHostInGroup(userId, groupId);
+    if (!isUserHasRoleHostInGroup) {
+      throw new ForbiddenException(ErrorCode.A009);
+    }
+    const group = await this.findById(groupId);
+    for (const email of dto.email) {
+      await this.mailService.inviteMemberToGroup(email, group.id, group.name);
+    }
```
Comment (LeTatTuan) [inline src/modules/group/group.service.ts:81]: - Dùng Promise.all chớ đừng lặp để goi bất đồng bộ m, cho hắn chạy song song á

File: src/modules/group/group.service.ts [hunk 4]
```diff
@@ -29,20 +37,77 @@ export class GroupService {
...
+
+  async joinRoom(userId: Uuid, groupId: Uuid) {
+    const group = await this.findById(groupId);
+    const isUserInGroup = await this.memberInGroupService.isUserInGroup(
+      userId,
+      groupId,
+    );
+    if (isUserInGroup) {
+      throw new ForbiddenException(ErrorCode.A014);
+    }
```
Comment (LeTatTuan) [inline src/modules/group/group.service.ts:92]: - 409 hoặc 400 chớ sao lại 403 m

File: src/modules/group/member-in-group.service.ts [hunk 5]
```diff
@@ -1,23 +1,48 @@
...
   constructor(
     private readonly memberInGroupRepository: MemberInGroupRepository,
   ) {}
 
-  async addMemberToGroup(userId: Uuid, roleInGroup: RoleInGroup, group: GroupEntity) {
+  async addMemberToGroup(
+    userId: Uuid,
+    roleInGroup: RoleInGroup,
+    group: GroupEntity,
+  ) {
```
Comment (LeTatTuan) [inline src/modules/group/member-in-group.service.ts:18]: - Mấy cái ni truyền object dto cho hắn đi m

File: src/modules/group/group.controller.ts (hunk 2 above)
Comment (duchuyyyy) [reply src/modules/group/group.controller.ts:95]: okm

File: src/modules/group/group.service.ts (hunk 4 above)
Comment (duchuyyyy) [reply src/modules/group/group.service.ts:92]: k có ở trong nhóm đó mà lại xem nội dung trong group thì Forbidden chứ nhể

File: src/modules/group/group.service.ts (hunk 3 above)
Comment (duchuyyyy) [reply src/modules/group/group.service.ts:81]: ok

File: src/modules/group/group.controller.ts (hunk 1 above)
Comment (duchuyyyy) [reply src/modules/group/group.controller.ts:83]: 2 params mấy có cần để object k m

**Pull request merged**